book/src/proptest-derive/errors.md

```diff
@@ -0,0 +1,631 @@
+```
+
+## E0018
+
+This error occurs if an unknown modifier is passed in `#[proptest(..)]`.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(frobnicate = "true")]
+struct Foo(u32);
+```
+
+Please see the [modifiers reference](modifiers.md) to see what modifiers are
+available.
+
+## E0019
+
+This error happens if anything extra is passed to `#[proptest(no_params)]`.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(no_params = "true")]
+struct Foo(u32);
+```
+
+`no_params` takes no configuration. The correct form is simply
+`#[proptest(no_params)]`.
+
+## E0020
+
+This error happens if anything extra is passed to `#[proptest(skip)]`.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+enum Foo {
+    V1,
+    #[proptest(skip = "yes")]
+    V2,
+}
+```
+
+`skip` takes no configuration. The correct form is simply `#[proptest(skip)]`.
+
+## E0021
+
+This error happens if `#[proptest(weight = <integer>)]` is passed an invalid
+integer or passed nothing at all.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+enum Foo {
+    #[proptest(weight)]
+    V1,
+    #[proptest(weight = heavy)]
+    V2,
+}
+```
+
+The only acceptable form is `#[proptest(weight = <integer>)]`, where
+`<integer>` is either an integer literal which fits in a `u32` or the same but
+enclosed in quotation marks.
+
+## E0022
+
+This error occurs if both `#[proptest(no_params)]` and `#[proptest(params = "type")]`
+are applied to the same item.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(no_params, params = "u8")]
+struct Foo(u32);
+```
+
+One attribute or the other must be picked depending on desired effect.
+
+## E0023
+
+This error happens if an invalid `#[proptest(params = "type")]` attribute is
+applied to an item.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(params = "Vec<u8")] // Note missing '>'
+struct Foo(u32);
+```
+
+There are a few different ways to get this error:
+
+- Pass nothing at all. E.g., `#[proptest(params)]`.
+
+- Pass something other than a string as the value. E.g.,
+  `#[proptest(params = 42)]`.
+
+- Pass a malformed type in the string, as in the example above. Note that type
+  syntax is limited to what the `syn` crate can understand. If you encounter a
+  valid type that is rejected here, you can work around it by defining a type
+  alias (e.g., `type MyParams = TypeThatSynCantParse;`). In such a case, please
+  consider [filing an issue](https://github.com/altsysrq/proptest/issues) as
+  well.
+
+## E0024
+
+This error happens if an invalid `#[proptest ..]` attribute is applied using a
+syntax the `proptest-derive` crate is not prepared to handle.
+
+Exactly what conditions can produce this error vary by Rust version.
+
+## E0025
+
+This error happens if both `#[proptest(strategy = "expr")]` and
+`#[proptest(value = "expr")]` are applied to the same item.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    #[proptest(value = "42", strategy = "Just(56)")]
```

Comment:
```suggestion
    #[proptest(value = "42", strategy = "Foo { bar: 56 }")]
```

Reply:
How is `Foo { bar: 56 }` a valid argument for `strategy` here?

Reply:
Ignore me; I was being silly here!